packages/ui/src/partials/w3m-account-button/index.ts

```diff
@@ -1,16 +1,100 @@
-import { ClientCtrl } from '@web3modal/core'
+import { AccountCtrl, ClientCtrl, ConnectModalCtrl, RouterCtrl } from '@web3modal/core'
+import type { FetchEnsAvatarOpts, GetBalanceOpts } from '@web3modal/ethereum'
 import { html } from 'lit'
-import { customElement } from 'lit/decorators.js'
+import { customElement, state } from 'lit/decorators.js'
+import '../../components/w3m-text'
+import '../../components/w3m-zorb-ens-image'
+
+import { CHEVRON, ETH_LOGO } from '../../utils/Svgs'
 import { global } from '../../utils/Theme'
 import ThemedElement from '../../utils/ThemedElement'
+import { formatAddress, roundBalance } from '../../utils/UiHelpers'
+import styles, { dynamicStyles } from './styles'
 
 @customElement('w3m-account-button')
 export class W3mAccountButton extends ThemedElement {
-  public static styles = [global]
+  public static styles = [global, styles]
+
+  // -- state & properties ------------------------------------------- //
+  @state() private address = ''
+  @state() private balance = ''
+  @state() private ens = ''
+
+  // -- lifecycle ---------------------------------------------------- //
+  public constructor() {
+    super()
+    this.getAccounts()
```

Comment:
This one is gonna be tricky, but I'll try to outline proper setps best I can:

1. Ideally we should have all of this data (acc, ens and balance) available here before the component is mounted. This can be done inside `chains/ethereum/src/utilities` `onClientConnected` function. And should also happen in `Promise.all`

2. Doing it in the above way also means that we probably need to move all `ConnectModalCtrl.closeModal` calls to `onClientConnected` as well, so that it closes after all the data is ready

3. Inside w3m-account-button component you would then need to not only access the data, but also subscribe to it (account can change if user changes it on their mobile wallet for example). You can see example of subscribing and unsubscribing to state in component like `w3m-modal-router`